services/scan/src/db_client.test.ts

```diff
@@ -0,0 +1,65 @@
+import * as fs from 'fs-extra';
+import * as tmp from 'tmp';
+import { DbClient } from './db_client';
+
+test('file database client', async () => {
+  const dbFile = tmp.fileSync();
+  const client = await DbClient.fileClient(dbFile.name);
+
+  await client.reset();
+  await fs.access(dbFile.name);
+
+  expect(client.getDatabasePath()).toBe(dbFile.name);
+  expect(client.isMemoryDatabase()).toBe(false);
+
+  await client.exec(
+    'create table if not exists muppets (name varchar(255) unique not null)'
+  );
+  await client.run('insert into muppets (name) values (?)', 'Kermit');
+  await client.run('insert into muppets (name) values (?)', 'Fozzie');
+
+  const backupDbFile = tmp.fileSync();
+  await client.backup(backupDbFile.name);
+
+  const clientForBackup = await DbClient.fileClient(backupDbFile.name);
+  expect(await clientForBackup.all('select * from muppets')).toEqual([
+    { name: 'Kermit' },
+    { name: 'Fozzie' },
+  ]);
+
+  await client.destroy();
+  await expect(fs.access(dbFile.name)).rejects.toThrowError('ENOENT');
+});
+
+test('memory database client', async () => {
+  const client = await DbClient.memoryClient();
+
+  await client.reset();
+
+  expect(client.getDatabasePath()).toEqual(':memory:');
+  expect(client.isMemoryDatabase()).toBe(true);
+
+  await client.destroy();
+});
+
+test('read/write', async () => {
+  const client = await DbClient.memoryClient();
+
+  await client.exec(
+    'create table if not exists muppets (name varchar(255) unique not null)'
+  );
+  expect(await client.all('select * from muppets')).toEqual([]);
+  expect(await client.one('select * from muppets')).toBeUndefined();
+
+  await client.run('insert into muppets (name) values (?)', 'Kermit');
+  await client.run('insert into muppets (name) values (?)', 'Fozzie');
+
+  expect(await client.all('select * from muppets')).toEqual([
+    { name: 'Kermit' },
+    { name: 'Fozzie' },
+  ]);
+  expect(await client.one('select * from muppets')).toEqual({ name: 'Kermit' });
+  expect(
+    await client.one('select * from muppets where name != ?', 'Kermit')
+  ).toEqual({ name: 'Fozzie' });
+});
```

Comment:
I checked the coverage and it looks like there are still a number of uncovered lines in db_client. Might be a good time to address them